api/v1alpha1/envoyproxy_types.go

```diff
@@ -122,6 +122,9 @@ type EnvoyProxyKubernetesProvider struct {
 	// are applied.
 	//
 	// +optional
+	// +kubebuilder:validation:XValidation:message="allocateLoadBalancerNodePorts can only be set for LoadBalancer type",rule="!has(self.allocateLoadBalancerNodePorts) || self.type == 'LoadBalancer'"
+	// +kubebuilder:validation:XValidation:message="loadBalancerIP can only be set for LoadBalancer type",rule="!has(self.loadBalancerIP) || self.type == 'LoadBalancer'"
+	// +kubebuilder:validation:XValidation:message="loadBalancerIP must be a valid IPv4 address",rule="!has(self.loadBalancerIP) || self.loadBalancerIP.matches(r\"^(((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\\.|$)){4})\")"
```

Comment:
this's hard to read, but I cannot find better solution right now.
Hope one day, there's ip lib from kubernetest just like [url](https://kubernetes.io/docs/reference/using-api/cel/#kubernetes-url-library).

Reply:
Unfortunately, this is just kind of the nature of CEL for more complex logic.